src/EditorFeatures/CSharpTest/Completion/CompletionProviders/AwaitCompletionProviderTests.cs

```diff
@@ -279,5 +279,231 @@ public async Task TestAwaitInLock_TopLevel()
         {
             await VerifyKeywordAsync("lock($$", LanguageVersion.CSharp9);
         }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnTask()
+        {
+            await VerifyKeywordAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task someTask)
+  {
+    someTask.$$
+  }
+}
+", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnValueTask()
+        {
+            var valueTaskAssembly = typeof(ValueTask).Assembly.Location;
+            await VerifyItemExistsAsync(@$"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"" LanguageVersion=""{LanguageVersion.CSharp9.ToDisplayString()}"">
```

Comment:
why C#9?

Reply:
No particular reason. Is the version optional?

Reply:
yup.  it's only needed if you're validating some specific behavior for a particular version.